### PR TITLE
Component: ColorPicker

### PR DESCRIPTION
**Pull request by zzmingo**
Add a component for color picking.

It's based on the very repo: https://github.com/casesandberg/react-color/

### CODE REVIEW

File: src/editor/mainEditor/material/logic/view/BasicMaterialView.ts [hunk 1]
```diff
@@ -3,3 +3,8 @@ import { setColor as setColorBuss } from "../bussiness/BasicMaterialBuss";
 export const setColor = (colorStr:string) => {
     setColorBuss(colorStr);
 };
+
+export const getColor = () => {
+    // I guess I uneed this from bottom layer
+    return "#FFFFFF"
```
Comment (zzmingo) [inline src/editor/mainEditor/material/logic/view/BasicMaterialView.ts]: I need an interface to get color from bottom layer.

File: src/ui/containers/App.tsx [hunk 2]
```diff
@@ -7,10 +7,18 @@ interface IProps{
...
+}
+
+class App extends React.Component<IProps,IState>{
 
     constructor(props:IProps){
         super(props);
+
+        this.state = {
+            testColor: "#FFFFFF"
+        }
```
Comment (yyc-git) [inline src/ui/containers/App.tsx]: remove testColor? it's not used.
remove all change in App.tsx

File: src/editor/mainEditor/material/ui/BasicMaterial.tsx [hunk 3]
```diff
@@ -1,14 +1,33 @@
 import * as React from "react";
+import ColorPicker, { ColorPickerType } from '../../../components/ColorPicker'
```
Comment (yyc-git) [inline src/editor/mainEditor/material/ui/BasicMaterial.tsx]: move ColorPicker to  src/editor/mainEditor/material/ui/component/

File: src/editor/mainEditor/material/logic/view/BasicMaterialView.ts (hunk 1 above)
Comment (yyc-git) [reply src/editor/mainEditor/material/logic/view/BasicMaterialView.ts]: ok, i will add it

**Comment by yyc-git** [comment]
you can add test case in test/test_ui/, refer to Translation.spec.tsx

File: src/editor/mainEditor/material/ui/BasicMaterial.tsx (hunk 3 above)
Comment (zzmingo) [reply src/editor/mainEditor/material/ui/BasicMaterial.tsx]: ColorPicker can use not only material color picking but also such as camera clear color, lighting color, so I made it there

Comment (yyc-git) [reply src/editor/mainEditor/material/ui/BasicMaterial.tsx]: yes, i know your idea, but now it's only used by material component.
if in the future it'll be used by other component, then move it at that time.

Comment (zzmingo) [reply src/editor/mainEditor/material/ui/BasicMaterial.tsx]: sure, you got it.

**Comment by zzmingo** [comment]
I updated, How to re-trigger the codecov.io ?

**Comment by yyc-git** [comment]
wo will see it

File: src/editor/mainEditor/material/components/ColorPicker.tsx [hunk 4]
```diff
@@ -1,8 +1,7 @@
 import * as React from "react";
-import { connect } from "react-redux";
 import { SketchPicker, BlockPicker, ChromePicker, CirclePicker, CompactPicker, HuePicker, MaterialPicker, SliderPicker, SwatchesPicker } from 'react-color';
 import reactCSS from 'reactcss';
-import { string2rgb, reverseRGB, rgb2hex, hex2string } from '../../utils/colorUtil';
+import { string2rgb, reverseRGB, rgb2hex, hex2string } from '../../../../utils/colorUtil';
```
Comment (yyc-git) [inline src/editor/mainEditor/material/components/ColorPicker.tsx]: refactor: material/components/ColorPicker.tsx should move to material/ui/components/

**Comment by yyc-git** [comment]
the codecov seems to be newest already.
https://codecov.io/gh/DreamForeast/Wonder-Editor/pull/1/commits

the reason of that the rate don't change  is that the rate is affected by "gulp testEditor"e, not by "npm run ui".

File: src/editor/mainEditor/material/components/ColorPicker.tsx (hunk 4 above)
Comment (zzmingo) [reply src/editor/mainEditor/material/components/ColorPicker.tsx]: resolved

**Pull request merged**